products/views.py

```diff
@@ -550,3 +561,110 @@ def delete_banner(request, pk):
         banner_data.delete()
         return redirect('banner')
     return render(request, 'product/banner/delete_confirm.html', {'banner': banner_data})
+
+
+@login_required
+def view_sections(request):
+    """ to display all the sections """
+
+    if not request.user.is_superuser:
+        raise Http404
+
+    sections = Section.objects.all().order_by('order')
+    search = request.GET.get('search')
+    if search is None:
+        search = ""
+    if search:
+        sections = Section.objects.filter(Q(name__icontains=search) | Q(order__icontains=search)
+                                          | Q(is_active__icontains=search))
+    else:
+        sections = sections
+
+    page = Paginator(sections, 3)
+    page_number = request.GET.get('page')
+    try:
+        page_obj = page.get_page(page_number)
+    except PageNotAnInteger:
+        page_obj = page.page(1)
+    except EmptyPage:
+        page_obj = page.page(page.num_pages)
+
+    context = {
+        'page_obj': page_obj,
+        'heading': AdminPortalHeadings.SECTIONS,
+        'search': search
+    }
+    return render(request, 'section/view_section.html', context)
+
+
+@login_required
+def update_section(request, pk):
+    """ To update section details """
+
+    if not request.user.is_superuser:
+        raise Http404
```

Comment:
move this to the common class or decorator.

Reply:
Done